### PR TITLE
First Assessment Page

### PR DESCRIPTION
**Pull request by silva-00**
Create a dropdown to show all roles

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -14,7 +14,9 @@
     "format": "prettier --write src/"
   },
   "dependencies": {
+    "20": "^3.1.9",
```
Comment (vitorlofonseca) [inline package.json:17]: What is it?

File: src/components/Divider.vue
Comment (vitorlofonseca) [inline src/components/Divider.vue:1]: You don't need this component, this is not being used in Figma

File: src/components/WelcomeItem.vue
Comment (vitorlofonseca) [inline src/components/WelcomeItem.vue:1]: What is this component? Is it being used?

File: src/stores/knowledges/actions.ts
Comment (vitorlofonseca) [inline src/stores/knowledges/actions.ts:1]: You don't need this. The only thing related to state management you should have, is a getter to get all the Roles in the database. Just this

File: src/stores/roles/state.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import type { Role } from '@/domain/Role'
+import { type Ref, ref } from 'vue'
+
+const roles: Ref<Role[]> = ref([])
+const quizState = {
```
Comment (vitorlofonseca) [inline src/stores/roles/state.ts:5]: You don't need this

File: src/views/FirstAssessmentPage.vue [hunk 3]
```diff
@@ -0,0 +1,72 @@
+<template>
+  <div class="container">
```
Comment (vitorlofonseca) [inline src/views/FirstAssessmentPage.vue:2]: Classes are CapitalCasing. Can you change the classes in this file to CapitalCasing?

File: src/views/FirstAssessmentPage.vue [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+
+onMounted(async () => {
+  await rolesStore.fetchRoles()
+})
+
+const localSelectedRole = ref(quizState.quiz.selectedRole)
+
+const handleRoleChange = (role: string) => {
+  try {
+    quizActions.setSelectedRole(role)
```
Comment (vitorlofonseca) [inline src/views/FirstAssessmentPage.vue:41]: You just need to do this in this method. Nothing more. Why you need localSelectedRole?

File: src/views/FirstAssessmentPage.vue [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+
+const rolesStore = useRolesStore()
+
+onMounted(async () => {
+  await rolesStore.fetchRoles()
+})
+
+const localSelectedRole = ref(quizState.quiz.selectedRole)
+
+const handleRoleChange = (role: string) => {
```
Comment (vitorlofonseca) [inline src/views/FirstAssessmentPage.vue:39]:
```suggestion
const handleRoleChange = (role: Role) => {
```

File: src/stores/knowledges/index.ts [hunk 6]
```diff
@@ -0,0 +1,15 @@
+import { defineStore } from 'pinia'
+import { removeKnowledge, saveKnowledge, editKnowledge, fetchKnowledges } from './actions'
+import { getKnowledge } from './getters'
+import { knowledges } from './state'
+
+export const useKnowledgeStore = defineStore('knowledges', () => {
+  return {
+    knowledges,
+    getKnowledge,
+    removeKnowledge,
```
Comment (fromgonie) [inline src/stores/knowledges/index.ts:10]: do you need all of this functions?

File: src/stores/roles/actions.ts [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+  if (getRoles?.value?.length > 0) {
+    return
+  }
+
+  const fetchedRoles = await get<Role[]>('/roles')
+
+  setRoles(fetchedRoles)
+}
+
+async function addRole(role: Role): Promise<void> {
```
Comment (fromgonie) [inline src/stores/roles/actions.ts:23]: do u need this methods?

File: src/domain/Level.ts
Comment (fromgonie) [inline src/domain/Level.ts:1]: do you need this file? youre not using levels.

**Review by fromgonie — CHANGES_REQUESTED** [review]
good job martin!

**Review by vascolusitan — APPROVED** [review]

File: src/components/Divider.vue
Comment (vitorlofonseca) [inline src/components/Divider.vue:1]: Do you need this component?

File: src/stores/quiz/actions.ts [hunk 8]
```diff
@@ -13,4 +15,8 @@ async function getQuizResult(quizAnswer: QuizAnswers) {
   return await post('/calculate-score', quizAnswer)
 }
 
-export { initializeQuizAnswers, getQuizResult }
+async function setSelectedRole(role: Role) {
+  quizAnswer.value.roleId = Number(role.id)
```
Comment (vitorlofonseca) [inline src/stores/quiz/actions.ts:19]: Do you need this Number()?

File: src/stores/quiz/getters.ts [hunk 9]
```diff
@@ -0,0 +1,5 @@
+import { quizAnswer } from './state'
+
+const quizGetters = {
```
Comment (vitorlofonseca) [inline src/stores/quiz/getters.ts:3]: A getter should always return something, and usually is a computed. Try to replicate the behaviour in other getters in admin project

**Review by vitorlofonseca — APPROVED** [review]

**Review by vascolusitan — APPROVED** [review]

**Review by fromgonie — APPROVED** [review]

**Pull request merged**